Gems/PhysX/Code/Tests/PhysXColliderComponentModeTests.cpp

```diff
@@ -386,4 +389,52 @@ namespace UnitTest
         PhysX::ColliderComponentModeRequestBus::BroadcastResult(subMode, &PhysX::ColliderComponentModeRequests::GetCurrentMode);
         EXPECT_EQ(PhysX::ColliderComponentModeRequests::SubMode::Dimensions, subMode);
     }
+
+    using PhysXColliderComponentModeManipulatorTest =
+        UnitTest::IndirectCallManipulatorViewportInteractionFixtureMixin<PhysXColliderComponentModeTest>;
+
+    TEST_F(PhysXColliderComponentModeManipulatorTest, AssetScaleManipulatorsScaleInCorrectDirection)
+    {
+        auto colliderEntity = CreateColliderComponent();
+        colliderEntity->FindComponent<TestColliderComponentMode>()->SetShapeType(Physics::ShapeType::PhysicsAsset);
+        colliderEntity->FindComponent<TestColliderComponentMode>()->SetAssetScale(AZ::Vector3::CreateOne());
+        EnterComponentMode<TestColliderComponentMode>();
+        PhysX::ColliderComponentModeRequestBus::Broadcast(&PhysX::ColliderComponentModeRequests::SetCurrentMode,
+            PhysX::ColliderComponentModeRequests::SubMode::Dimensions);
+
+        // position the camera so the X axis manipulator will be flipped
+        AzFramework::SetCameraTransform(
+            m_cameraState,
+            AZ::Transform::CreateFromQuaternionAndTranslation(
+                AZ::Quaternion::CreateRotationZ(-AZ::Constants::QuarterPi), AZ::Vector3(-5.0f, -5.0f, 0.0f)));
+
+        // select a point in world space slightly displaced from the position of the entity in the negative x direction
+        // in order to grab the X manipulator
+        const float x = 0.1f;
+        const float xDelta = 0.1f;
+        const AZ::Vector3 worldStart(-x, 0.0f, 0.0f);
+
+        // position in world space to drag to
+        const AZ::Vector3 worldEnd(-(x + xDelta), 0.0f, 0.0f);
+
+        const auto screenStart = AzFramework::WorldToScreen(worldStart, m_cameraState);
+        const auto screenEnd = AzFramework::WorldToScreen(worldEnd, m_cameraState);
+
+        m_actionDispatcher
+            ->CameraState(m_cameraState)
+            // move the mouse to interact with the x scale manipulator
+            ->MousePosition(screenStart)
+            // drag to move the manipulator
+            ->MouseLButtonDown()
+            ->MousePosition(screenEnd)
+            ->MouseLButtonUp();
+
+        const auto WorldToScreenMultiplier = 1.0f / AzToolsFramework::CalculateScreenToWorldMultiplier(worldStart, m_cameraState);
```

Comment:
variable name should start with lower case: `WorldToScreenMultiplier` -> `worldToScreenMultiplier`

Any chance that `CalculateScreenToWorldMultiplier` returns 0.0f causing a division by 0?

Reply:
Not with that camera state. 

The only way to get 0.0f would be to have the point you're looking at exactly in the plane of the camera (it's not here) _and_ set the near plane distance to 0.0 _and_ not have any floating point precision stuff taking you away from exactly 0. 

If that did happen it would break the code in the manipulator as well which also divides by it. I'll check with hultonha whether that needs to be guarded against in the manipulator code, although I think it's an extreme edge case, and I'm not sure you'd even be able to interact with the manipulator in that case because it might not be visible on the screen.